src/python/pants/backend/native/subsystems/native_build_step_settings_base.py

```diff
@@ -4,37 +4,114 @@
 
 from __future__ import absolute_import, division, print_function, unicode_literals
 
+from pants.backend.native.subsystems.native_build_settings import NativeBuildSettings
 from pants.backend.native.subsystems.utils.mirrored_target_option_mixin import \
   MirroredTargetOptionMixin
 from pants.subsystem.subsystem import Subsystem
+from pants.util.memo import memoized_property
 
 
 class NativeBuildStepSettingsBase(Subsystem, MirroredTargetOptionMixin):
 
+  options_scope = 'native-build-step-settings-base'
```

Comment:
I think that this class is supposed to be abstract, and then mixed in to a language specific subsystem? If so, it shouldn't need an `options_scope` to be defined, because the subclass will handle that.

Reply:
This is here to allow for base options sets that can be shared between C and CPP.